src/pages/index.js

```diff
@@ -36,58 +47,165 @@ const modalImageContainer = document.querySelector(
   ".modal__container--preview"
 );
 
+const popupWithDelete = new PopupWithDelete("#card-delete-modal");
+
 const profileFormValidator = new FormValidator(validationSettings, profileForm);
 
 const addCardFormValidator = new FormValidator(
   validationSettings,
   addModalForm
 );
 
+const profilePictureModalValidator = new FormValidator(
+  validationSettings,
+  profileAvatarForm
+);
+const popupWithImage = new PopupWithImage("#image-preview-modal");
+
+const api = new Api("https://around-api.en.tripleten-services.com/v1", {
+  authorization: "e2ed982e-7073-428a-82c0-3445ee97b908",
+  "Content-Type": "application/json",
+});
+
+const createCard = (data) => {
+  return new Card(
+    data,
+    "#card-template",
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
+  ).getElement();
+};
+
+const userInfo = new UserInfo({
+  name: ".profile__title",
+  about: ".profile__description",
+  link: ".profile__image",
+});
+
+const profileAvatarModal = new PopupWithForm(
+  "#profile-avatar-modal",
+  (formData) => {
+    const avatar = formData.avatar; // Get the avatar URL from formData
+
+    return api
+      .updatingProfilePic(avatar) // Pass the avatar directly
+      .then(({ avatar }) => {
+        userInfo.setAvatar(avatar); // Update avatar image in the DOM
+        const submitButton = document.querySelector("#avatar-modal-submit-btn");
+        submitButton.disabled = true;
```

Comment:
```js

        const submitButton = document.querySelector("#avatar-modal-submit-btn");
        submitButton.disabled = true;
```

Please, delete it because you disable the button on opening, so it's redundant here


And inside `PopupWithForm`, you enable the button in `finally`, so here it doesn't have any effect